LIMS2DB/classes.py

```diff
@@ -754,6 +756,17 @@ def get_library_preps(self, sample):
                             if not inp_artifact:
                                 self.log.error("Multiple copies of the same sample {0} found in step {0},  None of them is routed. Skipping the libprep ".format(sample.name, agrlibval.luid))
                                 continue
+                        except NoResultFound:
+                            #for the case of finished Libraries
+                            query = "select art.* from artifact art \
+                                inner join artifact_sample_map asm on  art.artifactid=asm.artifactid \
+                                inner join processiotracker piot on piot.inputartifactid=art.artifactid \
+                                inner join sample sa on sa.processid=asm.processid \
+                                where sa.processid = {sapid} and piot.processid = {agrid}".format(sapid=sample.processid, agrid=agrlv.processid)
+                            try:
+                                inp_artifact = self.session.query(Artifact).from_statement(text(query)).first()
+                            except:
```

Comment:
I'd prefer that you specify what exception you'd like to ignore, if that's possible? Ignoring all errors can be bad.